collection.go

```diff
@@ -236,19 +236,37 @@ func (c *Collection) AddDocument(ctx context.Context, doc Document) error {
 
 	// Persist the document
 	if c.persistDirectory != "" {
-		safeID := hash2hex(doc.ID)
-		docPath := filepath.Join(c.persistDirectory, safeID)
-		docPath += ".gob"
-		if c.compress {
-			docPath += ".gz"
-		}
+		docPath := c.getDocPath(doc.ID)
 		err := persist(docPath, doc, c.compress, "")
 		if err != nil {
-			return fmt.Errorf("couldn't persist document: %w", err)
+			return fmt.Errorf("couldn't persist document to %q: %w", docPath, err)
+		}
+	}
+
+	return nil
+}
+
+// RemoveDocument removes a document from the collection.
+func (c *Collection) RemoveDocument(_ context.Context, documentID string) error {
```

Comment:
WDYT about using the exact same naming as Chroma does here? As you wrote in the PR description they have [Delete()](https://github.com/chroma-core/chroma/blob/ce5c6b35e7ee7b2786eec6e3f5e7c821fb30d844/chromadb/api/models/Collection.py#L496-L501).

For "delete" vs "remove" we could just stick to their naming. And regarding the "...Document" suffix I'm thinking of `chromem-go`'s "Add" vs "AddDocument", where the former adheres more to Chroma, and the latter differs a bit. So when following Chroma's behavior here, we could also stay more close to their naming.

Another benefit is that we can later change the ID parameter to be variadic (`documentIDs ...string`) without a breaking change, and the "Delete" naming can stay as it's not tied to a single or multiple documents.

For the other parameters, if we want to add the `where` or `whereDocument` in the future, I think a breaking change is acceptable as long as we're < v1.0.0, and the README explicitly warns about it.